library/core/src/char/methods.rs

```diff
@@ -892,7 +892,8 @@ impl char {
     ///
     /// The general categories for numbers (`Nd` for decimal digits, `Nl` for letter-like numeric
     /// characters, and `No` for other numeric characters) are specified in the [Unicode Character
-    /// Database][ucd] [`UnicodeData.txt`].
+    /// Database][ucd] [`UnicodeData.txt`]. However ideographic numbers are considered alphabetic,
+    /// not numeric, please consider to use `is_ascii_digit` or `is_digit`.
```

Comment:
I don't think stringing these sentences together with a comma is the right thing to do. It could be interpreted that using those method would solve the "doesn't cover ideographic numbers" problem.

My impression from the issues and the meeting was that those should be separate statements:

a) This method doesn't cover everything that could be considered a number, e.g. ideographic numbers. If you want everything (defined by the unicode standard) including characters with overlapping purposes then you might want to use  a unicode library that exposes the appropriate character properties instead of looking at the categories
b) If you simply want to parse western arabic numerals (0-9) or base-N you'll want `is_ascii_digit` or `is_digit` instead

Reply:
For @the8472 's concern, I would suggest "This means that" instead of "However". Also it might be good to add a short explanation for what ideographic numbers are. Readers usually aren't unicode experts, know languages that have ideographic numbers, or have read #84056 .

Reply:
>My impression from the issues and the meeting was that those should be separate statements:

Yes, I think we should add a clarification inside the docs here, not much is an ideographic number, but when use and do not use the function.

I will try to introduce your suggestion in the PR. So, maybe we can bring this to the attention of the t-lib team in the meeting, and use the fresh argument in the head to make put a good solution here.

P.S: I will try to add also an example!